pkg/arvo/sys/vane/ames.hoon

```diff
@@ -283,17 +316,18 @@
       (lsh [3 +((add size.sndr-meta size.rcvr-meta))] content)
     ==
   =/  checksum  (end [0 20] (mug body))
-  =?  body  ?=(^ origin)  (mix u.origin (lsh [3 6] body))
+  =?  body  ?=(^ next)  (mix u.next (lsh [3 6] body))
   ::
   =/  header=@
     %+  can  0
-    :~  [3 reserved=0]
+    :~  [1 reserved=0]
+        [2 packet-type=0]
```

Comment:
We'll need at a bit to represent "is this a request?", i.e. not a response.  The `is-ames` bit should also be consolidated into the 2-bit `packet-type` field to save space, and of course this function should return these new values -- my inclination would be to add the fields to the `$packet` type.

This is a note as much to myself as to you:

Something to keep in mind with this sort of change to the packet format is that we should ensure packets at this version are discriminable from older packets.  In this case it's good enough just to look at the protocol version, but I was hoping each protocol could develop its own version to enable per-protocol updates later.  We could probably still do that, though, by looking at both the protocol version bits and the packet-type bits.

Reply:
What would the request/response bit be used for? My current implementation is request/response agnostic. I'm setting the next lane on every forwarded packet. Is this a bad idea?

Reply:
This makes sense to me. I can change the logic to treat requests and responses differently, but I'm not sure it's necessary.
My current relay logic works like this:
1. If we have a lane for sndr to rcvr in the pending map, use that.
2. If we are the direct sponsor of the rcvr, send to rcvr. (rcvr is in our sponsees map)
  --> otherwise, +try-next-sponsor up the sponsorship chain in case rcvr is our transitive sponsee
3. If we don't have an entry for the request in the pending map AND the rcvr isn't our direct or transitive sponsee, send to the rcvr's galaxy.

Regardless of request/response, +on-hear-packet stores [[rcvr sndr]] [lane (unit next-lane)] in the sponsorship map. I think this handles the routes correctly and maintains symmetry without needing to treat requests and responses differently.

Example:

request:  ~rovnys -> ~syl -> ~sipsyl -> ~ritpub-sipsyl -> ~mister-ritpub-sipsyl

~rovnys has no record of ~mister-ritpub-sipsyl in the pending requests map and is not a sponsor, so it sends the packet to the ~syl galaxy.
As the request is routed, each relay will enter [[~mister-ritpub-sipsyl ~rovnys] [previous-lane previous-previous-lane]] in its pending map. e.g. ~sypsyl will have [[~mister-ritpub-sipsyl ~rovnys] [(~syl's lane) (~rovnys's lane)]].

response: ~rovnys <- ~syl <- ~sipsyl <- ~ritpub-sipsyl <- ~mister-ritpub-sipsyl

On the way back, the relay's all have entries in their pending map for [~mister-ritpub-sipsyl ~rovnys], so they will send the packet to both lane and the next-lane.
As the request is routed, each relay will enter [[~rovnys ~mister-ritpub-sipsyl] [previous-lane previous-previous-lane]] in its pending map. e.g. ~sypsyl will have [[~rovnys ~mister-ritpub-sipsyl] [(~ritpub-sipsyl's lane) (~mister-ritpub-sipsyl's lane)].

Then, on the next request, the relays have entries in their pending map going both ways. Each relay has an entry in the pending map for [~rovnys ~mister-ritpub-sipsyl] and an entry for [~mister-ritpub-sipsyl ~rovnys]. i.e. after the first request, only the pending map is used to route packets. The result is that all of the relays then try to tighten the route in both directions.